doc/source/data/doc_code/quick_start.py

```diff
@@ -1,64 +1,119 @@
 
-ds.show(5)
-# -> {'col1': 0, 'col2': '0'}
-# -> {'col1': 1, 'col2': '1'}
-# -> {'col1': 2, 'col2': '2'}
-# -> {'col1': 3, 'col2': '3'}
-# -> {'col1': 4, 'col2': '4'}
+ds.show()
+# -> {'sepal.length': 5.1, 'sepal.width': 3.5, 'petal.length': 1.4, 'petal.width': 0.2, 'variety': 'Setosa'}
+# -> {'sepal.length': 4.9, 'sepal.width': 3.0, 'petal.length': 1.4, 'petal.width': 0.2, 'variety': 'Setosa'}
+# -> {'sepal.length': 4.7, 'sepal.width': 3.2, 'petal.length': 1.3, 'petal.width': 0.2, 'variety': 'Setosa'}
 
 ds.schema()
-# -> col1: int64
-# -> col2: string
+# -> sepal.length: double
+# -> sepal.width: double
+# -> petal.length: double
+# -> petal.width: double
+# -> variety: string
+# __create_from_python_end__
 
-# __data_setup_end__
 
-# __data_load_begin__
-import pandas as pd
-import dask.dataframe as dd
+# __create_from_files_begin__
+# Create from CSV.
+# Tip: "example://" is a convenient protocol to access python/ray/data/examples/data directory.
+ds = ray.data.read_csv("example://iris.csv")
+# -> Dataset(num_blocks=1, num_rows=150, schema={sepal.length: double, sepal.width: double, petal.length: double, petal.width: double, variety: string})
 
-# Create a Dataset from a list of Pandas DataFrame objects.
-pdf = pd.DataFrame({"one": [1, 2, 3], "two": ["a", "b", "c"]})
-ds = ray.data.from_pandas([pdf])
+# Create from Parquet.
+ds = ray.data.read_parquet("example://iris.parquet")
+# -> Dataset(num_blocks=1, num_rows=150, schema={sepal.length: double, sepal.width: double, petal.length: double, petal.width: double, variety: string})
+# __create_from_files_end__
 
-# Create a Dataset from a Dask-on-Ray DataFrame.
-dask_df = dd.from_pandas(pdf, npartitions=10)
-ds = ray.data.from_dask(dask_df)
-# __data_load_end__
 
+# __save_dataset_begin__
+# Write to Parquet files in /tmp/iris.
+ds.write_parquet("/tmp/iris")
+# -> /tmp/iris/data_000000.parquet
 
-# __data_transform_begin__
-ds = ray.data.range(10000)
-ds = ds.map(lambda x: x * 2)
-# -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1123.54it/s]
-# -> Dataset(num_blocks=200, num_rows=10000, schema=<class 'int'>)
-ds.take(5)
-# -> [0, 2, 4, 6, 8]
+# Use repartition to control the number of output files:
+ds.repartition(2).write_parquet("/tmp/iris2")
+# -> /tmp/iris2/data_000000.parquet
+# -> /tmp/iris2/data_000001.parquet
+# __save_dataset_end__
 
-ds.filter(lambda x: x > 5).take(5)
-# -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1859.63it/s]
-# -> [6, 8, 10, 12, 14]
 
-ds.flat_map(lambda x: [x, -x]).take(5)
-# -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1568.10it/s]
-# -> [0, 0, 2, -2, 4]
+# __data_transform_begin__
+import pyarrow as pa
+
+# Create 10 blocks for parallelism.
+ds = ds.repartition(10)
+
+# Filter out the Versicolor.
+def transform_batch(batch: pa.Table) -> pa.Table:
+    return batch.filter(pa.compute.equal(batch["variety"], "Versicolor"))
```

Comment:
I think that we might want to use a Pandas UDF here since users are far less likely to use Arrow for their UDFs.